draft-packages/title-block-zen/KaizenDraft/TitleBlockZen/SecondaryActions.tsx

```diff
@@ -41,28 +41,32 @@ const SecondaryActions = ({
   if (!secondaryActions && !secondaryOverflowMenuItems) return null
 
   const secondaryActionsAsToolbarItems = secondaryActions
-    ? secondaryActions.map(a => {
+    ? secondaryActions.map((a, i) => {
         if ("menuItems" in a) {
-          return (
-            <Menu
-              align="right"
-              button={
-                <Button
-                  secondary
-                  label={a.label}
-                  reversed={reversed}
-                  icon={chevronDownIcon}
-                  iconPosition="end"
-                />
-              }
-            >
-              <MenuContent>
-                {a.menuItems.map(menuItem => (
-                  <MenuItem {...menuItem} />
-                ))}
-              </MenuContent>
-            </Menu>
-          )
+          return {
+            key: `${i}`, // We shouldn't use an index here
```

Comment:
You left this comment which cleared this up for me:

> Unfortunately I needed to use the array index, against react best practices, because the menu items have no unique identifier. To fix this, we'd need to create a breaking change, which will force all consumers to add a unique key for each menu item.

IMHO it'd be more understandable if something similar was in these code comments too. I'm just trying to imagine what someone who hasn't seen this PR would think if they came across this comment saying "we shouldn't do this" but we're doing it anyway – having a bit of context on why we're doing it and what we should be doing in the future instead would be ideal, so anybody who wants to address this debt knows what to do.

I realise there are a few of them so we could even do something like have an explanation at the top of the file in a comment and all the comments direct the reader up there or something 🤷  Or alternatively we could create an issue and then link to that in the comment (I'm more than happy to do that if you like that idea).